src/Refinement/RefinementTransferOperators.jl

```diff
@@ -0,0 +1,166 @@
+
+"""
+  RefinementTransferOperator
+
+  Matrix-like operator providing a transfer between two `FESpaces`
+  defined in child-parent refined grids. 
+
+  The dofs are transfered by iteratively solving a projection problem 
+
+        (uh_to,vh_to)_to = (uh_from,vh_to)_Ω
+
+  where Ω is always chosen as the finest grid (i.e the RHS is
+  always integrated without loss of accuracy). 
+"""
+struct RefinementTransferOperator{T,A,B,C} <: AbstractMatrix{T}
```

Comment:
Why do you exactly need `<: AbstractMatrix{T}` ?  Many of the functions in the `AbstractMatrix{T}` abstract API are not being implemented.

Reply:
I think that we should use a more intention revealing name for this type. 

Something like `OrthogonalProjectionTransferOperator`. 

It is indeed a `TransferOperator`, but just a particular one from the different variants we may have.   

To-think: also on a related note, if we want to abstract this building block to give the whole framework more flexibility, shouldn't we consider an `Operator`-like abstract API, perhaps there is already one in Gridap ... to think ...

Reply:
BTW ... why do we need all these operators here, doesnt this belong to GridapSolvers.jl?

Reply:
Why should it belong to GridapSolvers? This machinery can be used outside of a solver, right? From what we discussed with @santiagobadia , AMR will eventually be in Gridap.jl, and we will need this in order to move FEFunctions around. So these operators should be here, not in GridapSolvers. Or at least this is how I understood things.

Reply:
Naming part resolved by [3db5b31](https://github.com/gridap/Gridap.jl/pull/838/commits/3db5b3116c952a7513223a84998adc847ea797fc)

Reply:
In our meeting today, we have discussed avoid the convenience of being able to write L2 projections when the Triangulation describing the domain of integration of the term on the left hand side and  that of right hand side do not match (but are nested, hierarchically defined). If we are able to achieve that with the high level API, from integration to assembly, we would write L2 projections (and also interpolations) among meshes just as we do it now when we have the same mesh. 

In such a case, the operators being implemented here wouldnt be actually neededs, just as we do not have a projection operation in Gridap within the same mesh. (it is just three lines of code)